docs/proposals/multi-platform.md

```diff
@@ -0,0 +1,556 @@
+
+#### Make kubelet automatically use the right pause image
+
+The `pause` is used for connecting containers into Pods. It's a binary that just sleeps forever.
+When Kubernetes starts up a Pod, it first starts a `pause` container, and let's all "real" containers join the same network by setting `--net=${pause_container_id}`.
+
+So in order to start Kubernetes Pods on any other architecture, an ever-sleeping image have to exist.
+
+Fortunately, `kubelet` has the `--pod-infra-container-image` option, and it has been used when running Kubernetes on other platforms.
+
+But relying on the deployment setup to specify the right image for the platform isn't great, the kubelet should be smarter than that.
+
+This specific problem has been fixed in [#23059](https://github.com/kubernetes/kubernetes/pull/23059).
+
+#### Vendored packages
+
+Here are two common problems that a vendored package might have when trying to add/update it:
+ - Including constants combined with build tags
+
+```go
+//+ build linux,amd64
+const AnAmd64OnlyConstant = 123
+```
+
+ - Relying on platform-specific syscalls (e.g. `syscall.Dup2`)
+
+If someone tries to add a dependency that doesn't satisfy these requirements; he/she should first fix the problem in the repository that is about to be added.
+When the dependency satisfies the Kubernetes requirements for compiling on all platforms, the PR can be merged.
+
+### kubectl should be released for all platforms that are relevant
+
+kubectl is released for more platforms than the proposed server platforms, if you want to check out an up-to-date list of them, [see here](../../hack/lib/golang.sh).
+
+kubectl is trivial to cross-compile, so if there's interest in adding a new platform for it, it may be as easy as appending the platform to the list linked above.
+
+### Addons
+
+Addons like dns, heapster and ingress play a big role in a working Kubernetes cluster, and we should aim to be able to deploy these addons on multiple platforms too.
+
+`kube-dns`, `dashboard` and `addon-manager` are the most important images, and they are already ported for multiple platforms.
+
+These addons should also be converted to multiple platforms:
+ - heapster, influxdb + grafana
+ - nginx-ingress
+ - elasticsearch, fluentd + kibana
+ - registry
+
+### Conflicts
+
+What should we do if there's a conflict between keeping e.g. `linux/ppc64le` builds vs merging a release blocker?
+
+In fact, we faced this problem while this proposal was being written; in [#25243](https://github.com/kubernetes/kubernetes/pull/25243). It is quite obvious that the release blocker is of higher priority.
+
+However, before temporarily [deactivating builds](https://github.com/kubernetes/kubernetes/commit/2c9b83f291e3e506acc3c08cd10652c255f86f79), the author of the breaking PR should first try to fix the problem. If it turns out being really hard to solve, builds for the affected platform may be deactivated and a P1 issue should be made to activate them again.
```

Comment:
Again, if we had CI that built and ran on all supported architectures, I will feel more comfortable.
I feel less comfortable claiming support on an arch without sufficient testing.